java-vtl-script/src/main/java/no/ssb/vtl/script/operations/AggregationOperation.java

```diff
@@ -76,35 +76,52 @@ protected DataStructure computeDataStructure() {
     @Override
     public Stream<DataPoint> getData() {
         DataStructure childStructure = getChild().getDataStructure();
+
+        Order.Builder orderBuilder = Order.create(childStructure);
+        groupBy.forEach(component -> orderBuilder.put(component, Order.Direction.ASC));
+        Order orderForSorting = orderBuilder.build();
+
+        Stream<DataPoint> data = getChild().getData(orderForSorting)
+                .orElseGet(() -> getChild().getData().sorted(orderForSorting));
+
+        return this.processStream(data);
+    }
+
+    @Override
+    public Optional<Stream<DataPoint>> getData(Order orders, Filtering filtering, Set<String> components) {
+        return getChild().getData(orders, filtering, components).map(this::processStream);
+    }
+
+    private Stream<DataPoint> processStream(Stream<DataPoint> data) {
+        DataStructure childStructure = getChild().getDataStructure();
         DataStructure structure = getDataStructure();
-    
-        Order.Builder builder = Order.create(childStructure);
-        groupBy.forEach(component -> builder.put(component, Order.Direction.ASC));
```

Comment:
The aggregation requires that the child returns its data sorted **first by all the identifiers except the one that is removed then with the identifier we are removing.** With this version of the code (which I wrote!), the order is fixed to ascending on all identifiers.

Example if we try to remove the **Gender** identifier: 

_Table in random order_

|*Country*|*Occupation*|*Gender*|Population|
|---|---|---|---|
|France|Teacher|Male|1|
|France|Barber|Male|2|
|Sweeden|Teacher|Male|3|
|France|Teacher|Female|4|
|Norway|Barber|Male|5|
|Norway|Teacher|Male|6|
|France|Barber|Female|7|
|Norway|Barber|Female|8|
|Sweeden|Barber|Male|9|
|Sweeden|Barber|Female|10|
|Norway|Teacher|Female|11|
|Sweeden|Teacher|Female|12|

_Sorted by **Country** then **Occupation**_

<table>
  <tr>
    <th>Country</th>
    <th>Occupation</th>
    <th>Gender</th>
    <th>Population</th>
  </tr>
  <tr>
    <th>France</th>
    <th>Teacher</th>
    <th>Male</th>
    <th rowspan="2">5</th>
  </tr>
  <tr>
    <td>France</td>
    <td>Teacher</td>
    <td>Female</td>
  </tr>
  <tr>
    <td>France</td>
    <td>Barber</td>
    <td>Male</td>
    <td rowspan="2">9</td>
  </tr>
  <tr>
    <td>France</td>
    <td>Barber</td>
    <td>Female</td>
  </tr>
  <tr>
    <td>Norway</td>
    <td>Teacher</td>
    <td>Male</td>
    <td rowspan="2">17</td>
  </tr>
  <tr>
    <td>Norway</td>
    <td>Teacher</td>
    <td>Female</td>
  </tr>
  <tr>
    <td>Norway</td>
    <td>Barber</td>
    <td>Male</td>
    <td rowspan="2">13</td>
  </tr>
  <tr>
    <td>Norway</td>
    <td>Barber</td>
    <td>Female</td>
  </tr>
  <tr>
    <td>Sweeden</td>
    <td>Teacher</td>
    <td>Male</td>
    <td rowspan="2">15</td>
  </tr>
  <tr>
    <td>Sweeden</td>
    <td>Teacher</td>
    <td>Female</td>
  </tr>
  <tr>
    <td>Sweeden</td>
    <td>Barber</td>
    <td>Male</td>
    <td rowspan="2"><br>19<br></td>
  </tr>
  <tr>
    <td>Sweeden</td>
    <td>Barber</td>
    <td>Female</td>
  </tr>
</table>

_Sorted by **Occupation** then by **Country**_

<table>
  <tr>
    <th>Country</th>
    <th>Occupation</th>
    <th>Gender</th>
    <th>Population</th>
  </tr>
  <tr>
    <td>France</td>
    <td>Barber</td>
    <td>Male</td>
    <td rowspan="2">9</td>
  </tr>
  <tr>
    <td>France</td>
    <td>Barber</td>
    <td>Female</td>
  </tr>
  <tr>
    <td>Norway</td>
    <td>Barber</td>
    <td>Male</td>
    <td rowspan="2">13</td>
  </tr>
  <tr>
    <td>Norway</td>
    <td>Barber</td>
    <td>Female</td>
  </tr>
  <tr>
    <td>Sweeden</td>
    <td>Barber</td>
    <td>Male</td>
    <td rowspan="2">19</td>
  </tr>
  <tr>
    <td>Sweeden</td>
    <td>Barber</td>
    <td>Female</td>
  </tr>
  <tr>
    <td>France</td>
    <td>Teacher</td>
    <td>Male</td>
    <td rowspan="2">5</td>
  </tr>
  <tr>
    <td>France</td>
    <td>Teacher</td>
    <td>Female</td>
  </tr>
  <tr>
    <td>Norway</td>
    <td>Teacher</td>
    <td>Male</td>
    <td rowspan="2">17</td>
  </tr>
  <tr>
    <td>Norway</td>
    <td>Teacher</td>
    <td>Female</td>
  </tr>
  <tr>
    <td>Sweeden</td>
    <td>Teacher</td>
    <td>Male</td>
    <td rowspan="2">15</td>
  </tr>
  <tr>
    <td>Sweeden</td>
    <td>Teacher</td>
    <td>Female</td>
  </tr>
</table>